### ISSUE TITLE
Wall and gate play balance

### ISSUE THREAD

**Issue by GeryonTom** [DayZMod/DayZ#771]
Players are now creating bases with no gates.  Since gates are the weakest point in base structures because they can be crowbarred or sledgehammered.  Plus the fact that gates required padlocks it's easier and more secure for bases now to be built with multiple layers of walls without gates.  The owner can dismantle the outer wall.  Get entrance to the inner wall, build up the outer wall and then dismantle the inner wall.  Play balance wise IMO I don't think this is what the devs intended.  The base becomes much more harder to breach since all parts of the it require rare explosives.  

For play balance gates or more importantly the lock should not be the weakest point.  Either remove the ability to break the lock or make it much more easier to breach walls (i.e. bring back hatcheting or single explosive).  This will prevent players from building self imposed prisons.

Alternatively make dismantling walls to not return any resources.  That way players can't simply build the wall back up.  However I can see players housing stashes of supplies inside the outer wall to counter this.

Since the walls are incredibly difficult to destroy, groups basically build fortress in any high traffic area.  In an sandbox this should be allowed.  However the inability for other motivated players to evict (i.e. like Epoch plot poles) or easily destroying the bases makes them simply troll bases.  We want to avoid house rules but virtually every town has these types of bases.  Since the risk of losing the base to someone else or having it destroyed is non-existent.  Again some balance so that the players can organically solve the in game challenges without having to resort to house rules.

**Comment by bchawke** [comment]
A long while back during 1.8.6 [I recommended allowing hatcheting to take down any wooden wall until it reached level 7, then requiring one satchel to take it down](https://github.com/DayZMod/DayZ/issues/573#issuecomment-153217865).  I know the current system is temporary and is actually a test of metal walls, but I agree with @GeryonTom that gates should be a bit harder to break into and wall should be a bit easier to tear down.

I like the suggestion of dismantling returning no resources.  If that's too harsh then perhaps dismantling a wall returns only half of the resources?

One thing I keep thinking about is what people's motivation for building a base is.  I think the idea of protecting gear/vehicles is flawed.  In my opinion there's no way to balance bases in vanilla mod so that they are hard to break into but not too hard, and hard to build but not too hard.  People will always be upset on both sides of the debate on how hard they are to break into.  I think that the mentality should be to build bases as a safe haven to cook food, warm up, rest at a tent, refuel, etc.  They should also be for staking claim on an area like Frogs in 434.  The minute they are made to difficult to penetrate that they're good for protecting gear/vehicles, vanilla mod has become Epoch and we might as well put safes in to permanently store gear.  The focus of balance should be less about _breaking in_ and more about _tearing down_.

**Comment by R4Z0R49** [comment]
@GeryonTom, what reports have you got from QF3 we made the gates harder to break into.

**Comment by Sportivoman** [comment]
I noticed the gates were harder to break into since the QF3, was playing Europa on off chance and went thru 6 crowbars and same amount of sledgehammers. Europa had QF3 before 434 and i was already established on there, Luke The Hero reported the same issue this morning breaking into a base 'Just bent 8 sledges and 7 crowbars and and still did not break open' 

I believe full upgraded walls / gates should be one satchel and small amount of explo crossbolts to knock down structure. Of late been seeing some very creative designs of bases all legit but it'd take a group an eternity to break down or break in. Its getting closer to being balanced that's for certain just some minor tweaks with the tear down system at set level and break in it'll be spot on.

**Comment by KPanicEKS** [comment]
I agree with all these points but also would suggest that base walls should
also not give you the option to dismantle from the outside. It will force
players to have a gate.

**Comment by GeryonTom** [comment]
@r4z0r49 I don't have any info on the gates being harder to break into with QF3.  That's from the other folks.  I didn't think anything changed from QF2 to 3 to be honest.  My only suggestion was for the gates to be harder to break into because players were opting not to build gates at all because they are considered the weakest point in bases.

I agree with M0tzilla maybe disable dismantling.  Something to discourage no gate bases.  Though that might be moot.  Slight topic change.  I'm hearing that players have figured out to glitch in and out to their bases without gates.  I'm going to do some testing on a test server to see what they might be doing.  I'll keep my findings private to you @R4Z0R49.

**Comment by R4Z0R49** [comment]
QF3 Changed tools breaking massively they went from a 4% chance to break upto 30% causing players to lose alot of tools during break-in.

**Comment by AndrewEscocia** [comment]
The take down and build again behind you seems like an exploit or at least bending the rules a bit. I would suggest a 50% return from taking down? it would be a quick and dirty fix .

If you look at historical castles/forts the gate would all ways be one of the strongest and most fortified parts of the construction so a case could be made for stronger gates and weaker walls  but this seems like another issue and not directly linked with ppl bricking up walls behind them.

**Comment by T-E-X** [comment]
@r4z0r49 Walls are becoming an issue in that they are so hard to destroy and can be build right in existing buildings. Players are blocking off entire hangers, ATC, ECT forcing admins to make rules; which then people object to and create work to enforce them. Please don't take this as negative feedback though, because it is awesome to have base building in the first place and I think balance can be achieved. Perhaps making it so walls can only be built a certain distance from existing structures, I don't like this option much because adding onto buildings with wooden walls is fun. The other option would be to add damage back in from other sources like the first time around, but maybe tweek it? What I think would be the best in the long run would be Metal walls that can only be taken down with satchel charges/explosive bolts that cannot be built anywhere near existing structures and wooden walls that can be built anywhere but are fairly easily destroyed. Thank you.

**Comment by CoftSock** [comment]
#removethebases

**Comment by rvett** [comment]
I fear that reducing the returned resources of dismantled structures won't solve the problem by far.

The balance of bases was brought up on Discord on October 21st, and we've postponed the discussion to a point in time when we would have more experience on how easy/hard it is to build up bases and to tear them down/break in/destroy walls etc. I went to bed that night thinking to myself: You know what? Let's see how hard it actually is (for me as a lone wolf, but a pretty experienced one at that) to do something about the people who build bases around loot spawn points or above highly frequented locations to have a safe haven from where they can snipe without any real risk. So I've actually spent the last _three weeks doing nothing but searching_ for explosives. I haven't had time to play every day, but perhaps every other or third day or so. I've been visiting the three new barracks at the Lopatino and Krasnostav airfields a bunch of times. I've also been running back and forth between Balota and Bor countless times to check as many deer stands as possible within the time I've been able to spare to play a computer game. I have climbed _a few hundreds of deer stands_. During these three weeks, I've found _one_ satchel charge and _three_ explosive bolts. 

I've had a closer look at the loot tables beforehand. If my math is right, one needs to visit 105 unlootet deer stands to find an explosive bolt and 24 unlootet new barracks to find a satchel on average. No other places to find explosives (please correct me if I missed something).

Of course I've had to stash those items. Unfortunately, one of my stashes that contained the satchel was compromised in the mean time, so I was left with the three explosive bolts. Still, I thought, that _has got to be enough_ to take down a base for good. So I went to somebody's base and fired the three bolts at a single spot (close to the edge between gate and wall, to hopefully damage both). Each was a full hit. _The wall doesn't have a scratch._ And there went my effort of three weeks worth of farming.

Guys, this _has_ to change. Bases are so out of balance right now. It took me much more time to collect all those explosives than I'm ever willing to invest again, and also way more than any casual or semi-casual DayZ player will be able to put into a single game. Sure, I was on my own. A whole team of players could have been faster and more effective. But if it takes a player who's willing to have a thorough look at the _source code_ of a game to optimize his play style to find the items he needs as efficiently as possible, a few weeks of dedicated farming and yet still he cannot do anything, not even take down a single piece of wall, I think it's just not balanced right.

If I had been able to take down at least one piece of wall, then who knows, I might have found a second and third layer behind it. And even if not, and I actually managed to break into a base, the owner wouldn't be having the slightest problem rebuilding those few pieces of wall within a matter of days at most, to have their base back and to keep beeing able to snipe out of that safe spot, or to keep having the loot that spawns inside for them exclusively.

And yes, I did try to break in with crowbars. But I've reached the point where it would have been more fun to spend my time playing a different game than looking for yet another pair of crowbars/sledgehammers.

I've expressed my concerns before about how tayloring the game to the veterans, making the strong ones even stronger and the weak ones weaker, is the wrong way to go in my opinion. With impregnable strongholds, this problem has been exacerbated once more in my eyes.

I've always loved how nothing was for keeps in DayZ. How everybody could get anywhere on the map. How there were no safe havens in the game. How you were at risk virtually anywhere. How everyone was able to take countermeasures against anything. This is all gone, at least for the lone wolfes. And certainly also for the returning casual players out there, who don't know _exactly_ where to look for explosives and such.

Alright, I've been complaining enough, now I would like to make a few suggestions to make my criticism constructive at least. :) I'm the kind of player who thinks that DayZ is much better without bases in the first place (I've thoroughly enjoyed 1.8.7), but if you guys like them and would like to keep them in the game -- fine. Here are a few ideas how I think play balance could be partially restored:

- Let explosives spawn in more places (bolts in farms and residentials, satchels in other static military spots rather than just the three new barracks).
- Let explosives spawn with higher probability (about a factor of at least five is needed from what I've experienced, if explosives are the way you indend to offer to take down bases).
- Increase the resource requirements for building structures, perhaps through a hierarchy of intermediate objects like wooden crates for tents for instance.
- Better even, introduce a special resource (glue, cement binder, some dedicated tool, ...) which is needed for any kind of construction and repairs, and make it much rarer than fishing poles and explosives. Make that special item breakable during construction.
- Make all walls and gates (except the highest levels perhaps) destructible with hatchets, such that it is actually a realistic endeavour to take them down this way
- Increase the break-in probability with crowbar/sledgehammer.
- Require less explosion damage to destroy walls and gates. One explosive of either kind should suffice to take ANY kind of player-built wall down.
- Remove raised stands from inner side of walls so pleople inside don't get the advantage of beging able to shoot out of bases without much risk.
- Restrict base building to areas away from existing buildings (might be hard to code, I know, I'm mentioning this here just for the sake of completeness).

To sum it up, bases should be at least as easy to destroy as to construct, or the map will just clutter up with strongholds or ruins of former ones.

There's a natural tendency of people to try to avoid loosing the upper hand they've been having over the weaker ones in life. After all, that's how politics and capitalism both work. So I fear that there's a lot at stake here for some folks who are active enough to take part in a development discussion on github. But for the sake of the game, which is supposed to be fun for more than just a few veterans, something needs to change.

Thanks for reading. I know it's a bit much text, but I'm feeling better now. :)

**Comment by R4Z0R49** [comment]
Thanks for the reply. We already know about a few of the issues you raised.

Explosives spawn in new vehicles also, The current damage system is for metal fences not wooden. But while we get the system tested we added it in. Break in is a random chance its pretty hard to balance anything random we have to look at other ways for this.

**Comment by KPanicEKS** [comment]
I stopped playing the mod (a game I played for about 4 years) due to the
poor implementation of base building.. Sorry I hate to be negative here but
bases in their current form are worse than they were before the removed
them.

**Comment by R4Z0R49** [comment]
Any real info to go with that?

**Comment by KPanicEKS** [comment]
I posted my concerns on discord a couple months ago but I'll post them here
again..

Essentially base walls shouldn't be impenetrable. As of a few months ago,
gates were the only base structures that took damage. AFAIK the base walls
themselves don't take damage and are apparently explosive proof. And they
can be removed from the outside which is one of their biggest flaws. Owners
simply use them as gates since they can dismantle them from the outside.
Why?
- My suggestion was to make it so base walls can only be dismantled from
  the inside.
- I also suggested that we should be able to craft explosive arrows being
  how rare they are.
- I also suggested that HE grenades and m203 should do damage to gates and
  walls.(why don't they currently)

If everyone is so concerned about their gear why not make the stashes
lockable? we have locks use them for other things beside base gates...

**Comment by R4Z0R49** [comment]
> AFAIK the base walls themselves don't take damage and are apparently explosive proof.

This is untrue the base walls take explosive damage from c4 there is a question if explosive arrows are working its yet to be confirmed we have had one video of them not working on a edited server but no other reports just now. I haven't had time to test yet due to long hours at work most ppl knew about.

As for the other suggestions they are being looked at.
- To dismantle base walls from the inside we need to edit the models to add a memory point. The reason this wasn't done is due to us still using a scripted selfActions system (a loop getting the object from the target your pointing at) this system don't really offer much play without change something we know we will be moving later to useractions. The new useractions will allow us to add this but will take time to add.

Your other two ideas have been brought up a few times release/preRelease to help balance bases. These will be addressed as and when we do patches seems they cant be done with a QF.

Our main concern with bases is making sure all the collision systems are working, any limits we have set in placement and seeing what exploits come up.

**Comment by thcNox** [comment]
We still get "feedback" from base owners saying it's too easy to break into their bases. Something which I don't really agree with. As a base owner who has a base that gets broken into and/or blown up on almost a daily basis and as someone who has broken into and blown up 4 bases (coincidentally I've only came across 4 bases worth the effort to break in) That's not to say it's without issues. Some that might be dealt with sooner rather than later but some are a bit out of scope for the near future.

My suggestions for fixes that can be done right now:

- Increase the maintenance requirement to 2 nails and one wooden plank (currently 1 nail and one wooden plank)

- Make walls level 3 and below take damage from everything including hatchets

- Deal with the issues facing explosive bolts (in my experience they don't work atm)

- Potentially make gp 25, m203 grenades, frag grenades be able to do damage to a wall, 5 to bring them down. Though I am extremely concerned with making such everyday items be able to take down bases as duping is a reality and satchels and explosive arrows are easily monitored by server admins while common grenades wouldn't be as manageable.

Largely I think most issues with bases being too hard to break into etc are "learn to play" related. If you claim to have farmed for months for explosives you may instead have spent a few hours in your nearest major city and farmed enough break in materials to break into any base on the server. As for blowing up bases atm the balance is a bit off between the effort to build  them and the effort required to blow them up but this is because while the metal base models are being prepared the metal base values are basically publically beta tested on the current wooden bases. Once the metal bases get implemented the effort to build goes up tenfold. That said I've managed to blow up a few bases myself and personally, as an attacker I feel satisfied with the effort required by me to blow up or otherwise gain access to a base.


.

**Comment by R4Z0R49** [comment]
None of the suggestions can be applied without a patch just an fyi atm.

**Comment by R4Z0R49** [comment]
Pushed QF5 to server owners it lowers the c4 to 1 and changes the chances for breakin.

Its there choice to add the changes. Some admins have already refused to add the changes.

**Comment by Landbergaren** [comment]
The server IntenZ will stay with QF4 since it have been proven to be a perfect balance for a server that have around 10-20 players playing at normal hours. Its been easy to monitor how bases have been working and we have seen a steady flow of teams breaking in to others bases. There have been wars between the massive bases and people have constantly been using all allowed methods to get in and been successful many of the times. The most common way is to break in through the gate. The second most common way is to use aircrafts to get people in to the base. Not many satchelcharges has been used since finding 2 requires so much work. But I still think its interesting that way since people will get through instantly, when using satchels instead of having to open maybe multiple of gates with tools.

The average experience is that you only need around 5-10 of each tool to be successful, at the most! Even though players have built the bases so you have to break in through many gates to get in, people have still gotten through. 

So we will pass on this one and check how the feedback comes along with the new changes.

**Comment by kooj12** [comment]
I'd like to see a change on the platform height on the walls, because now people are sitting crouch on them shooting players and you can barely see their head. Would it be possible to lower the height a bit so players have to stand up to be able to shoot and then it would be possible to see the players head fully or from neck and above

**Comment by skigoggles** [comment]
lots of good issues discussed. I'd say improvements would be dismantling only returns 50% AND I'd like to have a DZ_Safe_Build_Distance = NN (set by server) to be the distance you can build from anything of type "Building".  This would solve many issues. I'm thinking its in player_build, but maybe its object_build. I haven't had a chance to dive into this.

**Comment by bchawke** [comment]
I agree with @kooj12 100% on the height of the ledges.  I'm not a big critic of 3pp like some people are, but the current base wall ledge design promotes doing nothing but prone camping your base wall all day long, easily picking people off when they can barely see your head.

**Comment by jammerlich** [comment]
Why is there a dismantle option for walls in the first place? I can see the need during building, if you need to reposition something. Could walls be "finalized", after which point the dismantle option goes away? (but you can still upgrade if the wall begins to decay). An "un-finalized" wall would only last for a short time before decaying/removing (~a few resets, or 24 hours).

**Comment by Borges1091** [comment]
With the release of QF5 and QF6 I think it's too easy to gain access to bases. There are already a large number of squads with an abundant access to satchels and all it takes is one satchel to blow down any given wall. My base was breached the first day I created it. Once again not too long after that. With the crowbar/sledge probabilities reverted back to 30% and 20% walls should be balanced from explosives as well. 

Currently it takes 1 satchel to blow down a level 2 wall as well as a level 8 wall (whatever max is). It would be refreshing to see something like walls level 2-5 require 1 satchel to breach, and levels 6-8 requiring 2. This would give the player incentive to make their base better and be rewarded for the time they put into their base. There is absolutely no benefit in upgrading walls beyond level 2. Because of this bases have no real security. Please consider this, Having a base myself it's astonishing how frequently I log on to a blow down wall.

**Comment by devguy91** [comment]
My biggest problem with bases is that they are too intrusive to the game's overworld, for example, clans barricading entire portions of cities or taking up a large chunks of Chernarus's view with big walls. This is why I liked the old idea of underground bases, perhaps it could be an upgrade path from stashes, or otherwise discrete shelters used as a temporary safe haven for placing fires, shelter against the elements and crafting, similar to what bchawke said a while ago.
This idea has probably been discarded long ago, regardless, I'm sorry to say that in my opinion, the current implementation of big, almost indestructible bases filled with vehicles and a hoard of items are more akin to Overpoch or Rust's fantasy survival, which is against the original, hardcore zombie apocalypse simulation of Dayz.

Here's an image of a real life shelter by SurvivalLilly.
![maxresdefault](https://cloud.githubusercontent.com/assets/24664265/21740225/0c7e5e78-d518-11e6-99d5-f4fc1cb54ab6.jpg)

**Comment by rvett** [comment]
I agree with blackflarez that DayZ currently feels more like Epoch/Rust than the original, hardcore survival game. It's become a battle of strongholds, really. They're eveywhere.

I wonder if the current steep decline in player numbers has anything to do with how the clans are currently erecting forts around farm buildings etc. I woudn't be surprised. In fact I find myself losing interest in playing because of it.

**Comment by CoftSock** [comment]
> I wonder if the current steep decline in player numbers has anything to do with how the clans are currently erecting forts around farm buildings etc.

Absolutely not. Every update to the mod brings a rise and then month later it drops down, it has always been this way and the return of bases in the last update also meant the 'hype' from the update stayed longer. Its no coincidence that after an update drops the main servers go 60/60. I do agree however that building around structures/objects should not be happening.

**Comment by T-E-X** [comment]
Could I take this moment to propose again not allowing metal walls to be built near existing structures and making wooden walls ok to build near existing structures but easy to take out with melee. Please could anyone one the dev team give some feed back to this suggestion.
